lib/ecto/association.ex

```diff
@@ -115,6 +115,11 @@ defmodule Ecto.Association do
               {:assoc, t, {integer, atom}} | {:through, t, [atom]}
 
   @doc """
+  Preforms a preload operation (typically using `Ecto.Repo.Preloader.fetch_query/7`).
+  """
+  @callback preload(t, module, Ecto.Query.t, [atom], String.t, [Ecto.Schema.t], Keyword.t) :: Ecto.Schema.t | [Ecto.Schema.t] | Ecto.Query.t
```

Comment:
I hope this isn't too many things being passed into this callback. I was trying to keep it to a minimum but if you have any suggestions on how to remove some parameters I'm all ears...

Reply:
Can you name the parameters? IT will help understand what they are supposed to do, for example: `opts :: Ketword.t`.

Also, regarding the result type, doesn't it always returns a list of schemas?

Reply:
Yeah it does always return schemas. At one point I think I had it returning a query and forgot to update it.